packages/core/src/network.js

```diff
@@ -213,8 +220,11 @@ export class Network {
 
   // Called when a response has been received for a specific request. Associates the response with
   // the request data and adds a buffer method to fetch the response body when needed.
-  _handleResponseReceived = (session, event) => {
+  _handleResponseReceived = async (session, event) => {
     let { requestId, response } = event;
+    // await on requestWillBeSent
+    // no explicitly wait on requestWillBePaused as we implictly wait on it, since it manipulates the lifeCycle of request using Fetch module
+    await this.#requestsLifeCycleHandler.get(requestId).requestWillBeSent;
```

Comment:
```suggestion
    await this.#requestsLifeCycleHandler.get(requestId).requestWillBeSent;
```
[optional] add a method `lifecycle(requestId)` to shorten `this.#requestsLifeCycleHandler.get(requestId)`